include/mtxclient/crypto/client.hpp

```diff
@@ -85,13 +90,27 @@ unpickle(const std::string &pickled, const std::string &key)
 using OlmSessionPtr           = std::unique_ptr<OlmSession, OlmDeleter>;
 using OutboundGroupSessionPtr = std::unique_ptr<OlmOutboundGroupSession, OlmDeleter>;
 using InboundGroupSessionPtr  = std::unique_ptr<OlmInboundGroupSession, OlmDeleter>;
+using SASPtr                  = std::unique_ptr<OlmSAS, OlmDeleter>;
 
 struct GroupPlaintext
 {
         BinaryBuf data;
         uint32_t message_index;
 };
 
+struct SAS
+{
+        SAS();
+        std::string public_key();
+        void set_their_key(std::string their_public_key);
```

Comment:
Could you maybe write a short doc string for each of those explaining how they are meant to be used, i.e.

```cpp
//! After constructing a SAS object, you can set the key for the other party using this function. This needs to be done before generating bytes or calculating a MAC.
```

With that I'd say this should be ready to merge, so you may want to undraft it?

Reply:
Yeah I'll resolve that issue and add some explanation, maybe then if nothing else is left to do I'll undraft it